app/helpers/application_helper/title.rb

```diff
@@ -0,0 +1,120 @@
+module ApplicationHelper
+  module Title
+    def productized_title(title)
+      product_title + title
+    end
+
+    def product_title
+      # TODO: leave I18n until we have productization capability in gettext
+      I18n.t('product.name')
+    end
+
+    def page_title
+      @page_title.present? ? productized_title(@page_title) : title_from_layout(@layout)
+    end
+
+    # Derive the browser title text based on the layout value
+    def title_from_layout(layout)
+      title = product_title
+
+      return title if layout.blank?  # no layout, leave title alone
+
+      title + case layout
+              when "configuration"
+                _(": Configuration")
```

Comment:
@mzazrivec maybe in a separate PR, but these colons should be moved out from the gettext string as there is a high chance that the strings on the right side are already in the catalog.